org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -3649,12 +3649,25 @@ protected void consumeEnumConstantNoClassBody() {
 	final FieldDeclaration fieldDeclaration = (FieldDeclaration) this.astStack[this.astPtr];
 	fieldDeclaration.declarationEnd = endOfEnumConstant;
 	fieldDeclaration.declarationSourceEnd = endOfEnumConstant;
+
 	// initialize the starting position of the allocation expression
 	ASTNode initialization = fieldDeclaration.initialization;
 	if (initialization != null) {
 		initialization.sourceEnd = endOfEnumConstant;
 	}
+
+	// conditionally flush comments only if a comment starts inside the enum constant's source range
+	if (this.scanner.commentPtr >= 0) {
+		int sourceStart = fieldDeclaration.sourceStart;
+		int sourceEnd = fieldDeclaration.declarationSourceEnd;
```

Comment:
There's no need for sourceEnd. We can simply use endOfEnumConstant which is readily available.

Reply:
Done.